distributed/protocol/serialize.py

```diff
@@ -71,29 +74,111 @@ def pickle_loads(header, frames):
     return pickle.loads(x, buffers=buffers)
 
 
+def import_allowed_module(name):
+    if name in _cached_allowed_modules:
+        return _cached_allowed_modules[name]
+
+    # Check for non-ASCII characters
+    name = name.encode("ascii").decode()
```

Comment:
Out of curiosity, what prompted this check? Also, we should move this to the top of this function so we're consistent about what `name` we're putting into `_cached_allowed_modules` and checking with `if name in _cached_allowed_modules`.

Reply:
Paranoia, it is only there to make 100% sure that the string matching isn't returning false positives. Honestly, I don't think it is need but the overhead is negligible. 

No need to have it before the cache, since using the cache doesn't import code.